src/external/wannierio.jl

```diff
@@ -0,0 +1,693 @@
+We use here that:
+``u_{n(k + G_{\rm shift})}(r) = e^{-i G_{\rm shift} \cdot r} u_{nk}``.
+
+# Arguments
+- `basis`: the `PlaneWaveBasis`
+- `ψ`: the Bloch states, a vector of length `n_kpts`, each element is
+    a matrixof size `(n_G, n_bands)`. In the spin-polarized case, the
+    vector is of length `2*n_kpts`, and the first half is for spin-up,
+    the second half is for spin-down.
+- `kpb_k`: the kpoint indices of k+b, size: `(n_bvecs, n_kpts)`.
+    1st index: b-vector, 2nd index: kpoint, e.g., `kpb_k[ib, ik]` is the
+    corresponding kpoint index of k+b for `ib`-th b-vector at `ik`-th kpoint.
+- `kpb_G`: the shifting vectors of k+b, size: `(3, n_bvecs, n_kpts)`.
+    1st index: x,y,z component of b-vector,
+    2nd index: b-vector,
+    3rd index: shifting vector to get the shifted k-point `kpb_k[ib,ik]` to
+    the actual k+b point, thus the actual k+b is `kpb_k[ib,ik] + kpb_G[:,ib,ik]`.
+
+# Keyword arguments
+- `spin`: the spin channel to be used, either `1` or `2`.
+"""
+@timing function compute_mmn(
+    basis::PlaneWaveBasis, ψ::AbstractVector{<:AbstractMatrix{<:Complex}},
+    kpb_k::AbstractMatrix{<:Integer}, kpb_G::AbstractArray{<:Integer,3};
+    spin::Integer=1,
+)
+    n_bands = size(ψ[1], 2)
+    n_bvecs, n_kpts = size(kpb_k)
+    @assert size(kpb_G) == (3, n_bvecs, n_kpts)
+
+    kpts = krange_spin(basis, spin)
+    @assert length(kpts) == n_kpts
+    ik_shift = _get_ik_shift(basis, spin)
+
+    M = zeros(eltype(ψ[1]), n_bands, n_bands, n_bvecs, n_kpts)
+
+    for ik in kpts
+        k = basis.kpoints[ik]
+        ik0 = ik - ik_shift
+        for ib in 1:n_bvecs
+            ikpb0 = kpb_k[ib, ik0]
+            ikpb = ikpb0 + ik_shift
+            kpb = basis.kpoints[ikpb]
+            G_shift = kpb_G[:, ib, ik0]
+            # Search for common Fourier modes and their resp. indices in
+            # Bloch states k and kpb
+            # TODO Check if this can be improved using the G vector mapping in the kpoints
+            equivalent_G_vectors = [
+                (iGk, index_G_vectors(basis, kpb, Gk + G_shift))
+                for (iGk, Gk) in enumerate(G_vectors(basis, k))
+            ]
+            iGk = [eqG[1] for eqG in equivalent_G_vectors if !isnothing(eqG[2])]
+            iGkpb = [eqG[2] for eqG in equivalent_G_vectors if !isnothing(eqG[2])]
+
+            # Compute overlaps
+            for n in 1:n_bands
+                for m in 1:n_bands
+                    M[m, n, ib, ik0] = dot(ψ[ik][iGk, m], ψ[ikpb][iGkpb, n])
+                end
+            end
+        end
+    end
+    M
+end
+
+@doc raw"""
+Radial functions from solutions of Hydrogenic Schrödinger equation.
+
+Same as Wannier90 user guide Table 3.3.
+
+# Arguments
+- `r`: radial grid
+- `n`: principal quantum number
+- `α`: diffusivity, ``\frac{Z}/{a}`` where ``Z`` is the atomic number and
+    ``a`` is the Bohr radius.
+"""
+function radial_hydrogenic(r::AbstractVector{T}, n::Integer, α::Real=1.0) where {T<:Real}
```

Comment:
these hydrogenic things should live in common or maybe even in pspio. They sound like the could be useful for other things, too.